test/integration/DCAHubCompanion/position-migrator.spec.ts

```diff
@@ -89,10 +88,10 @@ describe('Position Migration', () => {
     // Send tokens from whales, to our users
     await distributeTokensToUsers();
 
-    // Approve swapper
-    await DCAHubSwapper.connect(msig).grantRole(await DCAHubSwapper.SWAP_EXECUTION_ROLE(), swapper.address);
-
     chainId = BigNumber.from((await ethers.provider.getNetwork()).chainId);
+
+    const legacyDCAHubSwapperFactory: LegacyDCASwapper__factory = await ethers.getContractFactory('LegacyDCASwapper');
+    legacyDCAHubSwapper = await legacyDCAHubSwapperFactory.deploy();
```

Comment:
Not entirely sure why we are deploying and using `Legacy` here instead of using the interface and the on-chain swapper? Am I missing something? 😬

Reply:
The on-chain swapper has a hardcoded hub. We want to execute swaps for all previous versions, so we needed to deploy a new contract for that. But since we are only using it here, it's under `mocks`